docs/framework/architecture/ui-library.md

```diff
@@ -439,6 +441,159 @@ dropdownView.bind( 'isEnabled' ).toMany( buttons, 'isEnabled',
+
+##### Header
+
+A header may consist of any combination of three elements:
+
+* an icon;
+* a title;
+* a "Close" button.
+
+The "Close" button is added automatically as long as you provide an icon or a title. The code below shows how to create a header with an icon and a title, but without a "Close" button.
+
+```js
+dialog.show( {
+	icon: pencilIcon,
+	title: 'My first dialog',
+	hasCloseButton: false
+} );
+```
+
+##### Contents
+
+This part can be a single [view](#views) or a collection of views. They will be displayed directly inside a body of a dialog.
+
+##### Actions
+
+The last piece of the dialog is the actions area, where the buttons are displayed. Their behaviour can be fully customized using the `onCreate()` and `onExecute()` callbacks. Below you can find an example of the configuration for the four custom buttons:
+
+* The "Ok" button that closes the dialog and has a custom CSS class set;
+* The "Set custom title" button that changes the dialog's title;
+* The "Drag detector" button that changes its state once the dialog was dragged by the user;
+* The "Cancel" button that closes the dialog.
+
+```js
+dialog.show( {
+	actionButtons: [
+		{
+			label: 'Ok',
+			class: 'ck-button-action',
+			withText: true,
+			onExecute: () => dialog.hide()
+		},
+		{
+			label: 'Set custom title',
+			withText: true,
+			onExecute: () => {
+				dialog.view!.headerView!.label = 'New title';
+			}
+		},
+		{
+			label: 'Drag detector'
+			withText: true,
+			onCreate: buttonView => {
+				buttonView.bind( 'isEnabled' ).to( dialog.view!, 'wasMoved', wasMoved => !wasMoved );
+			}
+		},
+		{
+			label: 'Cancel',
+			withText: true,
+			onExecute: () => dialog.hide()
+		}
+	]
+} );
+```
+
+#### Accessibility
+
+Dialogs provide full keyboard accessiblity. While a dialog is open, press the <kbd>Ctrl</kbd>+<kbd>F6</kbd> combination to move the focus between the editor and the dialog. It can also be closed at any time by pressing the <kbd>Esc</kbd> key. To navigate through the dialog, use the <kbd>Tab</kbd> and <kbd>Shift</kbd>+<kbd>Tab</kbd> keystrokes. The contents of the dialog are also available for screen readers.
+
+#### API
+
+The dialog's lifecycle (creating and destroying) is managed by the {@link module:ui/dialog/dialog Dialog plugin}. It provides two public methods: {@link module:ui/dialog/dialog~Dialog#show `show()`} and {@link module:ui/dialog/dialog~Dialog#hide `hide()`}. Both of them fire the respective events ({@link module:ui/dialog/dialog~DialogShowEvent `show`} and {@link module:ui/dialog/dialog~DialogHideEvent `hide`}), so it's possible to hook after or before them.
+
+##### `show()` method
+
+This method accepts a number of configuration options that allow to shape the structure and behaviour of the dialog.
+
```

Comment:
DialogDefinition is worth mentioning here.

Reply:
I'd say that an example is worth mentioning here.

In general, I am missing one, two or three nice examples. Maybe something simple, then the one with checkbox that I mentioned earlier.